plugins/content_distribution/providers/cross_kaltura/lib/batch/CrossKalturaDistributionEngine.php

```diff
@@ -228,8 +228,9 @@ protected function getEntryObjects(KalturaClient $client, $entryId, KalturaDistr
 	    
 	    // get entry
 	    KalturaLog::debug('Getting entry id ['.$entryId.']');
-	    $entry = $client->baseEntry->get($entryId);
-	    
+		KBatchBase::impersonate($this->distributionProfile->partnerId);
```

Comment:
Instead of all these internal impersonations, just impersonate before calling the getEntryObjects method and unimpersonate after it.